core/cli/src/lib.rs

```diff
@@ -501,7 +501,12 @@ where
 		config.telemetry_endpoints = Some(TelemetryEndpoints::new(cli.telemetry_endpoints));
 	}
 
-	config.force_authoring = cli.force_authoring;
+	// Imply forced authoring on --dev
```

Comment:
All of this can just be one line

`config.force_authoring = cli.shared_params.dev || cli.force_authoring`